packages/publish/test/publish.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Publish bot requires a minimum of Node 10 for the
+// runtime, we uses the lack of fs.promises to detect this.
+const { promises } = require('fs');
+if (!promises) {
+  console.warn('node 10 is required for publish bot');
+  process.exit(0);
+}
+
+import { Application } from 'probot';
+import handler from '../src/publish';
+
+import { resolve } from 'path';
+import { Probot } from 'probot';
+import snapshot from 'snap-shot-it';
+import nock from 'nock';
+import * as fs from 'fs';
+import { expect } from 'chai';
+
+nock.disableNetConnect();
+
+const fixturesPath = resolve(__dirname, '../../test/fixtures');
+
+interface Secret {
```

Comment:
Export `Secret` and `PublishConfig` to avoid having to duplicate the interface declarations?

Reply:
Right now, the way our bots are designed, they expect a single export which is the bot's handler; If I export the interfaces I get the following compile error:

```bash
: An export assignment cannot be used in a module with other exported elements
```

We could potentially pull the fixtures into a new file, if we find they're being updated frequently and the tests are getting more complex ... but I figured this is probably reasonable for now?